### ISSUE TITLE
About DataParallel and optimizer

### ISSUE THREAD

**Issue by zhjw0927** [open-mmlab/mmdetection#3372]
About DataParallel, I understand that module.forward() returns loss  that gather from multi gpus.
But, It is uncertain whether the optimizer is wrapped by DataParallel on the github.
Does optimizer need to be wrapped by DataParallel? 
Can you explain how the optimizer makes gradient updates and synchronizes across multi-gpus?
About issue, I have two thoughts:
1. on the gpu_ids[0], loss.backward(), then on multi-gpus optimizer update parameters.
2. loss.backward() on the gpu_ids[0] and optimizer.step() on gpu_ids[0],  then, optimizer broadcast updated parameters to multi-gpus?
which one is right?
Thank you for your answer!

**Comment by hellock** [comment]
You may read the source code of pytorch to learn about the optimizer and DataParallel. Optimizer is not wrapped.
BTW, We may remove the support for DataParallel in future releases and DistributedDataParallel is preferrable.
